### PR TITLE
31 v2 of logging

### PR DESCRIPTION
**Pull request by dapsavoie**
Due to the issues with the pipeline and merge conflicts, i recreated the code for [issue 6](https://github.com/civkit/civkit-node/issues/31) here. 

It contains a few edits and updates. Feel free to review and let me know what you think.

### CODE REVIEW

File: src/server.rs [hunk 1]
```diff
@@ -248,14 +251,25 @@ struct Cli {
...
+    let log_writer = util::init_logger(&data_dir)?;
 
+    // Initialize the logger and log the log file path
+    log::info!("Logging initialized. Log file located at: {:?}", data_dir.join("debug.log"));
+
+    let contents = fs::read_to_string("../../example-config.toml")
+        .expect("Something went wrong reading the file");
+
+    let config: Config = toml::from_str(&contents)
+        .expect("Could not deserialize the config file");
```
Comment (ariard) [inline src/server.rs:267]: “Could not deserialize the config file content” as here we succeed on the the read of the file at least

File: src/util.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+use tokio::sync::mpsc;
+use std::io::Write; 
+
+pub fn get_default_data_dir() -> PathBuf {
+    let home_dir = dirs::home_dir().expect("Home directory not found");
+
+    // Determine the platform-specific path
+    let platform_path = if cfg!(target_os = "linux") {
+        // Path for Linux (Debian/Ubuntu)
+        home_dir.join(".local/share/civkit-node")
```
Comment (ariard) [inline src/util.rs:16]: What do you think of `$HOME/.civkitd` as the canonical path for Linux :) ? Though `.local` is good too.

**Review by ariard — COMMENTED** [review]
Just few small comments, then happy to land it and carry on severity in other PRs.

**Comment by ariard** [comment]
Note you might have to rebase.

File: src/server.rs (hunk 1 above)
Comment (dapsavoie) [reply src/server.rs]: Updated in latest push

File: src/util.rs (hunk 2 above)
Comment (dapsavoie) [reply src/util.rs]: updated. makes sense to me to change.

**Comment by dapsavoie** [comment]
> Note you might have to rebase.

I think it should be fine? Seems we got around the issue.

**Comment by dapsavoie** [comment]
@ariard I think we can land? Thanks for patience on review with this :). Happy to make any changes if you see but will certainly add more in future branches.

**Comment by ariard** [comment]
Landed :)

**Comment by dapsavoie** [comment]
Its a pleasure :)

**Pull request merged**